### ISSUE TITLE
Global GetX onStart() is never called

### ISSUE THREAD

**Issue by michael-ottink** [jonataslaw/getx#412]
**Describe the bug**
OnStart thus OnInit / OnReady are never called for a global GetX controller

**To Reproduce**
Make a GetMaterial App
Make a HomePage widget
Add a GetX controller to the build method of the HomePage

**Expected behavior**
OnInit / OnReady to be called once

**Flutter Version:**
1.17.5

**Get Version:**
3.4.0

**Describe on which device you found the bug:**
Huawei p30 pro

**Minimal reproduce code**
```
void main() async {
  WidgetsFlutterBinding.ensureInitialized();
  runApp(GetMaterialApp(
    home: TestWidget(),
    smartManagement: SmartManagement.onlyBuilder,
  ));
}

class TestWidget extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: Center(
        child: GetX<MyController>(
          init: MyController(),
          builder: (MyController myController) {
            return Text(myController.test.value);
          },
        ),
      ),
    );
  }
}

class MyController extends GetxController {
  RxString test = ''.obs;

  @override
  void onInit() {
    print('onInit');
    super.onInit(); // this is not being printed
  }
}

```

**Output**

```
I/AwareBitmapCacher( 2495): init lrucache size: 2097152 pid=2495
I/flutter ( 2495): [GETX] INITIALIZED: If you need help, join our community support channels: https://tinyurl.com/y3cp88l3
I/flutter ( 2495): [GETX] GOING TO ROUTE /
```

**Comment by jonataslaw** [comment]
If you use "global:false" 
Does the error disappear?
This seems to be the only difference from your reproduction case to this one:
https://github.com/jonataslaw/getx/issues/407

**Comment by michael-ottink** [comment]
It is working properly with global set to false. But my GetX controller needs to work with as global. It used to before, but since the latest update my entire app is broken. I am going to try to downgrade for now.

**Comment by michael-ottink** [comment]
Can confirm on Get 3.3.0 this doesn't happen. Sadly #407 does happen on the version.

**Comment by jonataslaw** [comment]
Fix on 3.4.1
